src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpConstantExpectedAnalyzer.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.Analyzers.Performance;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Performance
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpConstantExpectedAnalyzer : ConstantExpectedAnalyzer
+    {
+        private static readonly CSharpDiagnosticHelper s_diagnosticHelper = new();
+        private static readonly IdentifierNameSyntax s_constantExpectedIdentifier = (IdentifierNameSyntax)SyntaxFactory.ParseName(ConstantExpected);
+        private static readonly IdentifierNameSyntax s_constantExpectedAttributeIdentifier = (IdentifierNameSyntax)SyntaxFactory.ParseName(ConstantExpectedAttribute);
+
+        protected override DiagnosticHelper Helper => s_diagnosticHelper;
+
+        protected override void RegisterAttributeSyntax(CompilationStartAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(context => OnAttributeNode(context), SyntaxKind.Attribute);
+        }
+        private void OnAttributeNode(SyntaxNodeAnalysisContext context)
+        {
+            var attributeSyntax = (AttributeSyntax)context.Node;
+            if (!attributeSyntax.Name.IsEquivalentTo(s_constantExpectedIdentifier) && !attributeSyntax.Name.IsEquivalentTo(s_constantExpectedAttributeIdentifier))
```

Comment:
This line is quite long. Don't we want to put it into a separate variable?
```suggestion
            var attributeName = attributeSyntax.Name;
            if (!attributeName.IsEquivalentTo(s_constantExpectedIdentifier) && !attributeName.IsEquivalentTo(s_constantExpectedAttributeIdentifier))
```